plugins/modules/postgresql_owner.py

```diff
@@ -292,7 +368,11 @@ def __is_owner(self):
                      "WHERE d.datname = %(obj_name)s "
                      "AND r.rolname = %(role)s")
 
-        elif self.obj_type == 'function':
+        elif self.obj_type in ('aggregate', 'function', 'routine', 'procedure'):
+            if self.obj_type == 'routine' and self.pg_version < 110000:
+                self.module.fail_json(msg="PostgreSQL version must be >= 11 for obj_type=routine.")
```

Comment:
```suggestion
            if self.obj_type == 'routine' and self.pg_version < 110000:
                self.module.fail_json(msg="PostgreSQL version must be >= 11 for obj_type=routine.")
```
This is just a FYI, no need to do it right now. Let's merge this PR if no objections.
In a future PR we could avoid the code repetitions. The ides is to create a matrix (but it maybe stupid and will not work but theoretically it should):
```
OBJ_TYPES_PG_VER_MAP = {
    'routine': 110000,
    ...
    'typeN': NUM,
}
```
Then in one place (maybe in the class in the initialization step) we could check it by calling a dedicated function containing:
```
if OBJ_TYPES_PG_VER_MAP['self.obj_type'] > self.pg_version:
    self.module.fail_json(msg="PostgreSQL version must be >= %s for obj_type=%s" % (OBJ_TYPES_PG_VER_MAP['self.obj_type'], self.obj_type))
```
For the old parameters present in all the version the collection supports we could just put `0`)

In `choices`, you can extract the keys using `OBJ_TYPES_PG_VER_MAP.keys()` IIRC the syntax

The idea for future PR